orca-queue-redis/src/main/kotlin/com/netflix/spinnaker/orca/q/redis/RedisQueue.kt

```diff
@@ -215,11 +228,11 @@ class RedisQueue(
         removeMessage(id)
       } else {
         try {
-          val message = mapper.readValue<Message>(json)
+          val message = convertToMessage(json, mapper)
           block.invoke(message)
         } catch(e: IOException) {
-          log.error("Failed to read message $id", e)
-          removeMessage(id)
+          log.error("Failed to read message $id, requeuing...", e)
+          requeueMessage(id)
```

Comment:
That should never happen unless we change the format. I don't think retrying will help